segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainer.java

```diff
@@ -58,4 +62,20 @@
         ArrayView segmentInfo = MetadataStore.SegmentInfo.recoveredSegment(streamSegmentName, length, isSealed);
         return metadataStore.createSegment(streamSegmentName, segmentInfo, new TimeoutTimer(TIMEOUT));
     }
+
+    @Override
+    public void copySegment(Storage storage, String sourceSegment, String targetSegment)
+            throws Exception {
+        storage.create(targetSegment, TIMEOUT);
+        val info = storage.getStreamSegmentInfo(sourceSegment, TIMEOUT);
+        int bytesToRead = (int) info.get().getLength();
+        int offset = 0;
+        while (bytesToRead > 0) {
+            byte[] buffer = new byte[Math.min(BUFFER_SIZE, bytesToRead)];
+            int size = storage.read(storage.openRead(sourceSegment).join(), offset, buffer, 0, buffer.length, TIMEOUT).join();
```

Comment:
Here, you invoke `openRead` on every loop call. This is highly inefficient. Invoke it once outside of the loop and then reuse the returned handle.

Reply:
also verify that `size > 0`. If it's 0, then you risk getting into an infinite loop.

Reply:
Done.